File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.py
class Solution:
    def maximumSubarraySum(self, nums: List[int], k: int) -> int:
        window = set()
        windowSum = 0     
        best = 0          
        left = 0

        for right in range(len(nums)):
          
            while nums[right] in window:
                window.remove(nums[left])
                windowSum -= nums[left]   
                left += 1

            # Add current element
            window.add(nums[right])
            windowSum += nums[right]

            # When window size reaches k, check and then slide
            if right - left + 1 == k:
                best = max(best, windowSum)
                window.remove(nums[left])
                windowSum -= nums[left]
                left += 1

        return best

File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        
        Set<Integer> window=new HashSet<>();
        long windowSum=0;
        long  subSub=0;
        int left=0;

        for(int right=0;right<nums.length;right++){
            while(window.contains(nums[right])){
                window.remove(nums[left]);
                windowSum-=nums[left++];
            
            }
            windowSum+=nums[right];
            window.add(nums[right]);
            if (right-left+1==k){
                subSub=Math.max(subSub,windowSum);
                window.remove(nums[left]);
                windowSum-=nums[left];
                left++;

            }
        }
        return subSub;
    }

}